src/main/java/com/mraof/minestuck/item/weapon/MusicPlayerItem.java

```diff
@@ -0,0 +1,234 @@
+	private final float volume;
+	private final float pitch;
+	
+	private static IItemHandler getItemStackHandlerMusicPlayer(ItemStack itemStack)
+	{
+		return itemStack.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY).orElseThrow(() ->
+				new IllegalArgumentException("Expected an item handler for the music player item, but " + itemStack + " does not expose an item handler."));
+	}
+	
+	private static IMusicPlaying getMusicPlaying(LivingEntity entity)
+	{
+		return entity.getCapability(MSCapabilities.MUSIC_PLAYING_CAPABILITY).orElseThrow(() ->
+				new IllegalArgumentException("Expected an music playing for this entity, but " + entity + " does not expose a music playing."));
+	}
+	
+	@Nullable
+	@Override
+	public ICapabilityProvider initCapabilities(ItemStack stack, @Nullable CompoundTag nbt)
+	{
+		return new MusicPlayerItemCapProvider();
+	}
+	
+	public MusicPlayerItem(Builder builder, Properties properties, float volume, float pitch)
+	{
+		super(builder, properties);
+		this.volume = volume;
+		this.pitch = pitch;
+		
+	}
+	
+	public MusicPlayerItem(Builder builder, Properties properties)
+	{
+		super(builder, properties);
+		this.volume = 4.0F;
+		this.pitch = 1.0F;
+	}
+	
+	@Override
+	public InteractionResultHolder<ItemStack> use(Level level, Player playerIn, InteractionHand handIn)
+	{
+		ItemStack musicPlayer = playerIn.getItemInHand(handIn);
+		
+		IItemHandler itemStackHandlerMusicPlayer = getItemStackHandlerMusicPlayer(musicPlayer);
+		IMusicPlaying musicPlayingCap = getMusicPlaying(playerIn);
+		
+		if(!level.isClientSide)
+		{
+			if(playerIn.isCrouching())
+			{
+				Item itemInMusicPlayer = itemStackHandlerMusicPlayer.getStackInSlot(0).getItem();
+				if(itemInMusicPlayer instanceof CassetteItem cassette)
+				{
+					MusicPlayerPacket packet;
+					if(musicPlayingCap.getCassetteType() == EnumCassetteType.NONE)
+					{
+						packet = MusicPlayerPacket.createPacket(playerIn, cassette.cassetteID, volume, pitch);
+						musicPlayingCap.setMusicPlaying(musicPlayer, cassette.cassetteID);
+					} else
+					{
+						packet = MusicPlayerPacket.createPacket(playerIn, EnumCassetteType.NONE, 0, 0);
+						musicPlayingCap.setMusicPlaying(ItemStack.EMPTY, EnumCassetteType.NONE);
+					}
+					MSPacketHandler.sendToTrackingAndSelf(packet, playerIn);
+				}
+			}
+			//open the GUI if right-clicked
+			else
+			{
+				MusicPlayerPacket packet = MusicPlayerPacket.createPacket(playerIn, EnumCassetteType.NONE, 0, 0);
+				musicPlayingCap.setMusicPlaying(ItemStack.EMPTY, EnumCassetteType.NONE);
+				MSPacketHandler.sendToTrackingAndSelf(packet, playerIn); //This will stop the music before opening the GUI
+				
+				NetworkHooks.openGui((ServerPlayer) playerIn, new SimpleMenuProvider((pContainerId, pInventory, pPlayer) ->
+						new CassetteContainerMenu(pContainerId, pInventory, itemStackHandlerMusicPlayer, musicPlayer),
+						new TranslatableComponent(TITLE)));
+			}
+			return InteractionResultHolder.success(musicPlayer);
```

Comment:
If any interaction would be a success, you should return a success regardless of the [logical side](https://forge.gemwire.uk/wiki/Sides/1.18). With that change, you should also be using `InteractionResultHolder.sidedSuccess()` instead of `InterectionResultHolder.success()`.